src/DiscGolfScorecardSecondary.java

```diff
@@ -136,10 +269,37 @@ public int totalScore() {
     public int holesPlayed() {
         int result = 0;
         for (int i = 0; i < this.length(); i++) {
-            DiscGolfScorecard.Hole hole = this.nextHole();
-            if (hole.strokes() > 0) {
+            if (this.currentHole().strokes() > 0) {
                 result++;
             }
+            this.advanceHole();
+        }
+        return result;
+    }
+
+    // CHECKSTYLE: ALLOW THIS METHOD TO BE OVERRIDEN
+    @Override
+    public int furthestHolePlayed() {
+        boolean found = false;
+        int currentHoleNumber = this.currentHoleNumber();
+        int i = 0;
+        while (!found && i < this.length()) {
+            //Calling abstract method due to simplicity
+            this.retreatHole();
```

Comment:
Am I allowed to call the abstract method this.retreatHole()? If not this method would be very annoying to implement.